code/src/pages/CastDetails.js

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from "react"
+import { useParams, useHistory } from "react-router-dom"
+
+import { CAST_URL } from '../utils/urls'
+import Loader from "../components/Loader"
+import CastProfile from "../components/CastProfile"
+import HomePath from "../components/HomePath"
+
+const CastDetails = () => {
+  const { id } = useParams()
+  const history = useHistory()
+  const [cast, setCast] = useState()
+
+  useEffect(() => {
+    fetch(CAST_URL(id))
+      .then(res => res.json())
+      .then(data => {
+        if (data.status_message) {
+          history.push("/notfound")
```

Comment:
This one is giving the same NotFound-page as the movie list, saying "This is not the movie you're looking for!" despite being on the cast section. We're guessing you were planning to write another text for the cast tho ;)